database/pipeline.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2022 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package database
+
+import (
+	"bytes"
+	"compress/zlib"
+	"database/sql"
+	"errors"
+	"io/ioutil"
+
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrEmptyPipelineNumber defines the error type when a
+	// Pipeline type has an empty Number field provided.
+	ErrEmptyPipelineNumber = errors.New("empty pipeline number provided")
+
+	// ErrEmptyPipelineRef defines the error type when a
+	// Pipeline type has an empty Ref field provided.
+	ErrEmptyPipelineRef = errors.New("empty pipeline ref provided")
+
+	// ErrEmptyPipelineRepoID defines the error type when a
+	// Pipeline type has an empty RepoID field provided.
+	ErrEmptyPipelineRepoID = errors.New("empty pipeline repo_id provided")
+
+	// ErrEmptyPipelineVersion defines the error type when a
+	// Pipeline type has an empty Version field provided.
+	ErrEmptyPipelineVersion = errors.New("empty pipeline version provided")
+)
+
+// Pipeline is the database representation of a pipeline.
+type Pipeline struct {
+	ID              sql.NullInt64  `sql:"id"`
+	RepoID          sql.NullInt64  `sql:"repo_id"`
+	Number          sql.NullInt32  `sql:"number"`
+	Flavor          sql.NullString `sql:"flavor"`
+	Platform        sql.NullString `sql:"platform"`
+	Ref             sql.NullString `sql:"ref"`
+	Type            sql.NullString `sql:"type"`
+	Version         sql.NullString `sql:"version"`
+	ExternalSecrets sql.NullBool   `sql:"external_secrets"`
+	InternalSecrets sql.NullBool   `sql:"internal_secrets"`
```

Comment:
Isn't `native` used elsewhere in the codebase?

Reply:
`native` is the secret driver (a.k.a. `engine`) used for managing internal secrets

https://go-vela.github.io/docs/tour/secrets/

https://go-vela.github.io/docs/reference/yaml/secrets/#the-engine-tag

However, the nomenclature attached to secrets fetched by the `native` driver has been `internal secrets`:

https://go-vela.github.io/docs/usage/examples/secrets_external/

https://go-vela.github.io/docs/usage/examples/secrets_internal/